js/modulestest/runtime.go

```diff
@@ -0,0 +1,51 @@
+// Package modulestest contains helpers to test js modules
+package modulestest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/eventloop"
+	"go.k6.io/k6/lib"
+)
+
+// Runtime is a helper struct that contains what is needed to run a (simple) module test
+type Runtime struct {
+	VU            *VU
+	EventLoop     *eventloop.EventLoop
+	CancelContext func()
+}
+
+// NewRuntime will create a new test runtime and will cancel the context on test/benchmark end
+func NewRuntime(t testing.TB) *Runtime {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	vu := &VU{
+		CtxField:     ctx,
+		RuntimeField: goja.New(),
+	}
+	vu.RuntimeField.SetFieldNameMapper(common.FieldNameMapper{})
+	eventloop := eventloop.New(vu)
+	vu.RegisterCallbackField = eventloop.RegisterCallback
+	result := &Runtime{
+		VU:            vu,
+		EventLoop:     eventloop,
+		CancelContext: cancel,
+	}
+	// let's cancel again in case it has changed
+	t.Cleanup(func() { result.CancelContext() })
+	return result
+}
+
+// InitContext will set the InitContext of the Runtime
+func (r *Runtime) InitContext(i *common.InitEnvironment) {
```

Comment:
```suggestion
func (r *Runtime) SetInitContext(i *common.InitEnvironment) {
```

Reply:
Will wait on that to do it with https://github.com/grafana/k6/pull/2602#discussion_r928558087 if we decide on anything

Reply:
I think the consensus there is to keep as-is so I think we can proceed to make a decision here.

Reply:
I kind of wonder if I shouldn't just initialize the InitEnv with some very basic values and then let the users directly customize it before calling `MoveToVUContext` :thinking: That will also make some of the usage easier :thinking: 

WDYT @oleiade

Reply:
I find `SetInitContext` somewhat more intuitive, as without it my future self would have to remember that I need to set a `common.InitEnvironment` in the `r.VU.InitEnvField`. Consider it a helper/documentation method 👍🏻

Reply:
What I meant @oleiade is that `r.VU.InitEnvField` *will* be set to some structure with some fields - most of them just initialized to not be nil (I can't really have a default `filesystems` for every test). And then you will only set the fields that you need (or the whole thing if you need to I guess)

Reply:
I made a PR with the removal of SetInitContext and having a default one that seems okay for ... all our current tests https://github.com/grafana/k6/pull/2621